libpromises/evalfunction.c

```diff
@@ -173,7 +173,18 @@ static FnCallResult FnFailure(void)
 
 static VarRef* ResolveAndQualifyVarName(const FnCall *fp, const char *varname)
 {
-    VarRef *ref = VarRefParse(varname);
+    VarRef *ref = NULL;
+    if (NULL != varname && IsVarList(varname))
+    {
+        char naked[CF_MAXVARSIZE] = "";
+        GetNaked(naked, varname);
```

Comment:
The documentation patch will have to be slightly updated if you agree with this change.